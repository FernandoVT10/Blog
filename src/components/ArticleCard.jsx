import Link from "next/link";

import "../styles/components/articlecard.scss";

export default ({ article }) => {
    return (
        <div className="article-card">
            <img
            src={`/img/articles/${article.cover}`}
            className="article-card__cover"
            alt={article.title} />

            <div className="article-card__details">
                <h3 className="article-card__title">{ article.title }</h3>

                <p className="article-card__description">{ article.description }</p>

                <Link href="/article/[id]" as={`/article/${article._id}`}>
                    <a className="article-card__link">
                        Continue Reading

                        <i className="fas fa-arrow-right ml-1" aria-hidden="true"></i>
                    </a>
                </Link>
            </div>
        </div>
    );
};