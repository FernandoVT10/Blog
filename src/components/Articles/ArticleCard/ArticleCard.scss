.article-card {
    width: 100%;
    background: var(--primary);
    color: var(--white);

    &__cover {
        object-fit: cover;
        width: 100%;
        height: 200px;
    }

    &__details {
        padding: 20px;
    }

    &__title {
        font-weight: bold;
        font-size: 1.25em;
    }

    &__description {
        font-weight: bold;
        font-size: 0.8em;
    }

    &__link {
        color: var(--white);
        font-weight: bold;
        font-size: 14px;
        transition: .3s;

        &:hover {
            text-decoration: none;
            color: var(--secondary);
        }
    }
}