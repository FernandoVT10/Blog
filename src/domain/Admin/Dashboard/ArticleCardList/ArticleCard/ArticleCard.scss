.dashboard-home-article-card {
    width: calc(50% - 40px);
    background: var(--primary);

    &:nth-child(1) { margin-right: 40px; }

    &__content {
        padding: 15px;
        padding-bottom: 0;
    }

    &__label {
        display: block;
        color: vaR(--white);
    }

    &__views {
        display: block;
        color: vaR(--white);
        font-weight: bold;
        font-size: 20px;

        &--loading {
            display: block;
            width: 100px;
            height: 30px;
            background: var(--black);
        }
    }

    &__cover {
        margin-top: 10px;
        object-fit: cover;
        width: 100%;
        height: 200px;

        &--loading {
            display: block;
            margin-top: 10px;
            width: 100%;
            height: 200px;
            background: var(--black);
        }
    }

    &__title {
        color: vaR(--white);
        font-weight: 600;
        font-size: 22px;
        margin: 20px 15px;

        &--loading {
            margin: 20px 15px;
            display: block;
            width: 80%;
            height: 35px;
            background: var(--black);
        }
    }

    &__footer {
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 15px;
        background: var(--black);

        &-text {
            cursor: pointer;
            color: var(--secondary);
            font-weight: bold;
        }
    }

    &__loader {
        color: var(--secondary);
    }

    @media screen and (max-width: 992px) {
        width: 100%;
        margin: 0;
        
        &:nth-child(1) { margin-bottom: 40px; }
    }
}