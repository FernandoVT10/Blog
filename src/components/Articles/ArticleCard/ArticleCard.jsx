import Link from "next/link";

import "./ArticleCard.scss";

const ArticleCard = ({ article }) => {
    return (
        <div className="article-card">
            <img
            src={`/img/articles/${article.cover}`}
            className="article-card__cover"
            alt={article.title}
            loading="lazy" />

            <div className="article-card__details">
                <h3 className="article-card__title">{ article.title }</h3>

                <p className="article-card__description">{ article.description }</p>

                <Link href="/article/[id]" as={`/article/${article._id}`}>
                    <a className="article-card__link">
                        Continue Reading

                        <i className="fas fa-arrow-right ml-1" aria-hidden="true"></i>
                    </a>
                </Link>
            </div>
        </div>
    );
}

export default ArticleCard;