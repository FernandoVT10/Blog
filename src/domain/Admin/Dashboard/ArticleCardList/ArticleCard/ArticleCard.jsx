import ArticleItem from "./ArticleItem";

import separateThousands from "../../../../../services/separateThousands";

import "./ArticleCard.scss";

export default (props) => {
    // This is the most viewed article or main article
    const getMainArticle = () => {
        const mainArticle = props.data[0];

        if(!mainArticle) {
            return (
                <div>
                    <span className="dashboard-home-article-card__views--loading"></span>
                    <span className="dashboard-home-article-card__cover--loading"></span>
                    <span className="dashboard-home-article-card__title--loading"></span>
                </div>
            );
        }

        return (
            <div>
                <span className="dashboard-home-article-card__views">
                    { separateThousands(mainArticle.views) }
                </span>

                <img
                src={`/img/articles/${mainArticle.cover}`}
                className="dashboard-home-article-card__cover"
                alt={ props.title } />

                <h3 className="dashboard-home-article-card__title">
                    { mainArticle.title }
                </h3>
            </div>
        );
    }

    const getFooter = () => {
        if(!props.loading) {
            return (
                <span className="dashboard-home-article-card__footer-text"
                onClick={() => props.showArticlesTable()}>
                    Show All Articles
                </span>
            );
        }

        return (
            <div
            className="dashboard-home-article-card__loader"
            role="status">
                <span className="spinner-border"></span>
            </div>
        );
    }

    return (
        <div className="dashboard-home-article-card">
            <div className="dashboard-home-article-card__content">
                <span className="dashboard-home-article-card__label">
                    { props.title }
                </span>

                { getMainArticle() }

                <div className="dashboard-home-articles__articles-list">
                    {props.data.map((article, index) => {
                        // if the index is 0 it means that it's the main article
                        // then we skip it
                        if(!index) {
                            return;
                        }

                        return <ArticleItem article={article} key={index}/>;
                    })}
                </div>
            </div>

            <div className="dashboard-home-article-card__footer">
                { getFooter() }
            </div>
        </div>
    );
}