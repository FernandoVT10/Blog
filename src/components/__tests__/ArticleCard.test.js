import renderer from "react-test-renderer";
import ArticleCard from "../ArticleCard/";

const ARTICLE_MOCK = {
    _id: 95,
    title: "Test title",
    cover: "test-image.jpg",
    description: "Test description"
};

test('Check if <ArticleCard/> renders correctly', () => {
    const component = renderer.create(
        <ArticleCard article={ARTICLE_MOCK}/>
    );

    expect(component.toJSON()).toMatchSnapshot();
});