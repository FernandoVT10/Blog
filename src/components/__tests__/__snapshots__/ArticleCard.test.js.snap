// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <ArticleCard/> renders correctly 1`] = `
<div
  className="article-card"
>
  <img
    alt="Test title"
    className="article-card__cover"
    src="/img/articles/test-image.jpg"
  />
  <div
    className="article-card__details"
  >
    <h3
      className="article-card__title"
    >
      Test title
    </h3>
    <p
      className="article-card__description"
    >
      Test description
    </p>
    <a
      className="article-card__link"
      href="/article/95"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Continue Reading
      <i
        aria-hidden="true"
        className="fas fa-arrow-right ml-1"
      />
    </a>
  </div>
</div>
`;
